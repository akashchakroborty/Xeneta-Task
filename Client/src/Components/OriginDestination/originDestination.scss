.originDestination {
  display: flex;
  align-items: center;
  div {
    margin: 1em;
  }
}
