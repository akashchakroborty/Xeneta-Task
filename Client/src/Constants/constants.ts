export const SELECT_ORIGIN = 'Select Origin';
export const SELECT_DESTINATION = 'Select Destination';
export const EMPTY_STRING = '';
export const DEFAULT_PORTS = [];
export const DEFAULT_RATES = [];
export const DEFAULT_LOADING = false;
