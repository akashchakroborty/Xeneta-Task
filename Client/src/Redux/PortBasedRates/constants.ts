export const SET_PORTS_SUCCESS = 'SET_PORTS_SUCCESS';
export const SET_PORTS_FAILURE = 'SET_PORTS_FAILURE';
export const GET_PORTS_LOADING = 'GET_PORTS_LOADING';
export const UPDATE_ORIGIN = 'UPDATE_ORIGIN';
export const UPDATE_DESTINATION = 'UPDATE_DESTINATION';
