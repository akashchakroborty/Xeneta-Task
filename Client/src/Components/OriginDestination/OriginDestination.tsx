import React from 'react';
import { SwapOutlined } from '@ant-design/icons';
import DropdownComponent from '../../CommonComponents/Dropdown/Dropdown.container';
import {
  EMPTY_STRING,
  GET_RATES,
  SELECT_DESTINATION,
  SELECT_ORIGIN,
} from '../../Constants/constants';
import Button from '../../CommonComponents/Button/Button';
import { RatesParam } from '../../Redux/Types/portBasedRatesTypes';

import './originDestination.scss';

export interface OriginDestinationProps {
  getRates: (props: RatesParam) => void;
  origin: string;
  destination: string;
}

const OriginDestination: React.FC<OriginDestinationProps> = ({ getRates, origin, destination }) => {
  const handleClick = () => {
    getRates({ origin, destination });
  };
  const isGetRatesButtonDisabled = origin === EMPTY_STRING || destination === EMPTY_STRING;
  return (
    <div className="originDestination">
      <DropdownComponent defaultOption={SELECT_ORIGIN} />
      <SwapOutlined className="icon" />
      <DropdownComponent defaultOption={SELECT_DESTINATION} />
      <Button
        type="primary"
        value={GET_RATES}
        onClick={handleClick}
        disabled={isGetRatesButtonDisabled}
      />
    </div>
  );
};

export default OriginDestination;
