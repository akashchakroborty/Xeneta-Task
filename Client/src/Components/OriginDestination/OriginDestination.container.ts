import { connect } from 'react-redux';
import { ThunkDispatch } from 'redux-thunk';
import { getRates } from '../../Redux/PortBasedRates/actionCreators';
import { selectDestination, selectOrigin } from '../../Redux/PortBasedRates/selector';
import { AppState } from '../../Redux/store';
import { PortBasedRatesActionTypes } from '../../Redux/Types/portBasedRatesActionTypes';
import { RatesParam } from '../../Redux/Types/portBasedRatesTypes';
import OriginDestination from './OriginDestination';

const mapStateToProps = (state: AppState) => ({
  origin: selectOrigin(state),
  destination: selectDestination(state),
});

const mapDispatchToProps = (dispatch: ThunkDispatch<AppState, void, PortBasedRatesActionTypes>) => {
  return {
    getRates: (props: RatesParam) => {
      // eslint-disable-next-line @typescript-eslint/no-floating-promises
      dispatch(getRates(props));
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(OriginDestination);
