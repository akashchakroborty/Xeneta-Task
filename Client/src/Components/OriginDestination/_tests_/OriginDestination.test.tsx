/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import OriginDestination, { OriginDestinationProps } from '../OriginDestination';
import { EMPTY_STRING } from '../../../Constants/constants';

describe('should test the OriginDestination Component', () => {
  const mockProps: OriginDestinationProps = {
    getRates: jest.fn(),
    origin: EMPTY_STRING,
    destination: EMPTY_STRING,
  };
  test('should render the Component', () => {
    const component = shallow(<OriginDestination {...mockProps} />);
    expect(component).toMatchSnapshot();
  });
  test('should test handleClick function', () => {
    const wrapper = shallow(<OriginDestination {...mockProps} />);
    wrapper.find('ButtonComponent').simulate('click');
    expect(mockProps.getRates).toHaveBeenCalled();
  });
  test('Button should be enabled with valid inputs', () => {
    const newMockProps: OriginDestinationProps = {
      ...mockProps,
      origin: 'mockOrigin',
      destination: 'mockDestination',
    };
    const wrapper = shallow(<OriginDestination {...newMockProps} />);
    expect(wrapper.props().children[3].props.disabled).toBe(false);
  });
});
