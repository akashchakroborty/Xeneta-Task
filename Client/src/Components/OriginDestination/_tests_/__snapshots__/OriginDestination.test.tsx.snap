// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the OriginDestination Component should render the Component 1`] = `
<div
  className="originDestination"
>
  <Connect(DropdownComponent)
    defaultOption="Select Origin"
  />
  <div>
    <ForwardRef(SwapOutlined)
      className="icon"
    />
  </div>
  <Connect(DropdownComponent)
    defaultOption="Select Destination"
  />
  <ButtonComponent
    disabled={true}
    onClick={[Function]}
    type="primary"
    value="Get Rates"
  />
</div>
`;
